src/v/utils/base64.cc

```diff
@@ -121,3 +121,24 @@ bytes base64url_to_bytes(std::string_view data) {
     // NOLINTNEXTLINE
     return {reinterpret_cast<bytes::const_pointer>(srv.data()), srv.size()};
 }
+
+iobuf base64_decode(const iobuf& buf) {
```

Comment:
* it looks like this is missing a test?

in the future, consider plucking this out into separate prs.

Reply:
> * it looks like this is missing a test?
Yep.

> in the future, consider plucking this out into separate prs.

Agreed.

Reply:
Added a test